test/e2e/util/util.go

```diff
@@ -757,6 +757,35 @@ func EnsureNetworkPolicies(t *testing.T, ctx context.Context, c crclient.Client,
 	})
 }
 
+// EnsureNodesRuntime ensures that all nodes in the NodePool have the expected runtime handlers.
+// This is only supported on 4.18+ when the default runtime is changed to crun.
+func EnsureNodesRuntime(t *testing.T, ctx context.Context, c crclient.Client, nodepool *hyperv1.NodePool) {
```

Comment:
nodepool in the signature seems unused

Reply:
Seems like you should filter nodes by NodePool in the list below, given that you're doing this when the given nodepool is ready, not necessarily all nodes.

Reply:
But all the nodes, independently of the nodepool should has those runtimeHandlers. I really matter the nodepool assigned to that node? If it matter I can add it :). just let me know.

Reply:
Ahh ok, makes sense. I will modify the tests in that way. btw it's already gated behind `Version18` tests :).